### PR TITLE
Huili

### PR DESCRIPTION
**Pull request by huilichen25**
Just added a new class which is able to measure the phonemic distance between two words

### CODE REVIEW

File: TapGameController/TapGameUtils/arpabet_mapping.csv [hunk 1]
```diff
@@ -1,6 +1,6 @@
-a,AA		wad,dot,odd
-b,B		bad
-c,AO		o in "or", au in "caught", aw in "awe"
+a,AA,		wad,dot,odd
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/arpabet_mapping.csv]: Good to keep consistency with line endings

File: TapGameController/TapGameUtils/wpsm.csv [hunk 2]
```diff
@@ -0,0 +1,40 @@
+,AA,AE,AH,AO,AW,AY,B,CH,D,DH,EH,ER,EY,F,G,HH,IH,IY,JH,K,L,M,N,NG,OW,OY,P,R,S,SH,T,TH,UH,UW,V,W,Y,Z,ZH
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/wpsm.csv]: What does this file do?

File: TapGameController/TapGameUtils/PronunciationResultsHandler.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
+import pronouncing
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/PronunciationResultsHandler.py]: Most of this file is already present in `TapGameUtils/PronunciationUtils.py`, can you take the parts of this file that you need and add them there?

File: TapGameController/TapGameUtils/PronunciationResultsHandler.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+		for phoneme in phonemes:
+			if phoneme in self.arpabet_map:
+				output+=self.arpabet_map[phoneme]
+			else:
+				print("phone ( "+phoneme+" ) does not exist in arpabet map")
+				break
+		# print(output)
+		return output
+
+	def measure_weighted_levenshtein_distance(self,word1,word2):
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/PronunciationResultsHandler.py]: Dependencies should either be included in the requirements.txt (if it's a pip install-able python package), explicitly stated in the README.rst (preferably with a one-line install for OS X and Ubuntu using homebrew or apt-get), or included in the repo itself (if it's not available via a package manager)

File: TapGameController/TapGameUtils/PronunciationResultsHandler.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
+import pronouncing
+import csv
+import os
+import pandas as pd
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/PronunciationResultsHandler.py]: Add external pip-installable dependencies to requirements.txt

Comment (huilichen25) [inline TapGameController/TapGameUtils/PronunciationResultsHandler.py]: added

File: TapGameController/TapGameUtils/PronunciationResultsHandler.py (hunk 4 above)
Comment (huilichen25) [inline TapGameController/TapGameUtils/PronunciationResultsHandler.py]: thanks. changed already

File: TapGameController/TapGameUtils/PronunciationUtils.py [hunk 6]
```diff
@@ -201,3 +203,65 @@ def get_phonetic_similarity_rep(self, word):
...
+
+        result=lev(word1.encode(),word2.encode(),substitute_costs=substitute_costs)
+        print(result)
+        
+        #normalize the levenshtein score by taking max(str1,str2)
+        denominator = max(len(word1),len(word2))
+        normalized_score = result / float(denominator)
+        print(normalized_score)
+        return normalized_score
+
```
Comment (SamSpaulding) [inline TapGameController/TapGameUtils/PronunciationUtils.py]: Good idea to add tests! The best place for them is in the top-level 'tests' folder. You could make a new file called 'test_pronunciation.py' or something like that. Unit testing is a bit of a new skill (one that even now, I'm not quite sure I have the hang of), but give it a go if you're interested and we can work on it together

**Pull request merged**